### ISSUE TITLE
bad_verification_code using github (redundant calls)

### ISSUE THREAD

**Issue by tacman** [knpuniversity/oauth2-client-bundle#213]
The access token when using github expires after the first use, so the example code fails.  

```php
public function checkCredentials($credentials, UserInterface $user): bool
    {
        return $this->fetchAccessToken($this->getGithubClient());
    }

  public function getUser($credentials, UserProviderInterface $userProvider)
    {
        /** @var GithubResourceOwner $githubUser */
        $githubUser = $this->getGithubClient()
            ->fetchUserFromToken($credentials);
   }
```

The problem is that you can't call BOTH fetchAccessToken and fetchUserFromToken, because the token expires after the first call and fails on the second (it took me way too long to figure this out...)

Things seem to work okay if I return true for checkCredentials(), but is that the right approach?

**Comment by weaverryan** [comment]
Yo Tac!

I think... you're describing a legitimate situation... but not quite with the right code. But I need your help to verify. The `fetchAccessToken()` method DOES use the one-time "code" to get the access token. You cannot repeat this again, that's 100% true. But `$this->getGithubClient()->fetchUserFromToken($credentials);` uses the re-usable access token to fetch the user information. This IS allowed.

Basically, the code above SHOULD work. BUT, what should *not* work is the very-similar `$this->getGithubClient()->fetchUser();` because that, internally, tries to "fetch the token" again.

So, are you sure you have the exact code above? Or are/were you possible using `->fetchUser()` instead? At one time, we *did* have the docs wrong and were showing that function.

Cheers!
